fxapom/pages/sign_in.py

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env python
+
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from base import Base
+
+from selenium.webdriver.common.by import By
+from selenium.webdriver.support.ui import WebDriverWait
+
+
+class SignIn(Base):
+
+    _fox_logo_locator = (By.ID, 'fox-logo')
+    _email_input_locator = (By.CSS_SELECTOR, '.input-row .email')
+    _next_button_locator = (By.ID, 'email-button')
+    _password_input_locator = (By.ID, 'password')
+    _sign_in_locator = (By.ID, 'submit-btn')
+
+    def __init__(self, testsetup):
+        Base.__init__(self, testsetup)
+
+        if len(self.selenium.window_handles) > 1:
+            self.popup = True
+            for handle in self.selenium.window_handles:
+                self.selenium.switch_to.window(handle)
+                if self.is_element_present(*self._fox_logo_locator):
```

Comment:
I tried that initially and the login screen used by Marketplace, which is the only suite currently using this tool, behaves strangely in that the title switches from "Firefox Accounts" to "Sign in ....", so there were timing issues. I tried a number of things and this was the most reliable one I came up with.